lib/gssapi/spnego/init_sec_context.c

```diff
@@ -567,9 +571,16 @@ spnego_reply(OM_uint32 * minor_status,
 				       &mech_output_token,
 				       &ctx->mech_flags,
 				       &ctx->mech_time_rec);
-	    if (GSS_ERROR(ret))
+	    if (GSS_ERROR(ret)) {
 		gss_mg_collect_error(ctx->selected_mech_type, ret, minor);
+	    }
```

Comment:
I think here you should `if (negState == incomplete && ret == GSS_S_COMPLETE && mech_output_token.length) negState = complete;`

Reply:
Then we may trample on `negState = requestMic`, I think?

Reply:
Of course we won't, sorry.

Reply:
I've applied your patch without the output token length check, remind me why it's needed?

Reply:
It turns out (after discussion offline with Nico) that the original patch was correct. negState refers to the state as viewed by the acceptor upon entering the function, further code in `spnego_reply()` relies on this and it should not be updated to the *current* state after the mechanism ISC completes.